addon/adapters/firebase.js

```diff
@@ -502,8 +502,10 @@ export default DS.Adapter.extend(Ember.Evented, {
   _getRef: function(type, id) {
     var ref = this._ref;
     var typeName = type;
-    if (type && type.typeKey) {
-      typeName = type.typeKey;
+
+    // Fall to typeKey for Ember Data < 1.0beta.18
+    if (type && (type.modelName || type.typeKey)) {
```

Comment:
Indeed it is - http://emberjs.com/blog/2015/05/21/ember-data-1-0-beta-18-released.html. `pathForType` seems to take care of most of the craziness. Added a camelize in the two other places.